spec/features/labels_controller_spec.rb

```diff
@@ -0,0 +1,133 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe 'LabelsController', js: true do
+  let(:admin) { create(:admin) }
+  let(:password) { attributes_for(:admin)[:password] }
+
+  before do
+    sign_in_with_js_driver(admin, password)
+  end
+
+  describe '#index' do
+    context 'when using name filter' do
+      let!(:matching_labels) { [create(:label, name: 'Loops'), create(:label, name: 'Java loops')] }
+      let!(:not_matching_labels) { [create(:label, name: 'label1'), create(:label, name: 'python for loop')] }
+
+      it 'filters by name correctly' do
+        visit(labels_path)
+        fill_in id: 'label-name-filter-input', with: 'loops'
+        find('input', id: 'label-name-filter-input').send_keys(:enter)
+        wait_for_ajax
+        matching_labels.each {|label| expect(page).to have_selector('.labels-table > tbody > tr > td > .task_label', text: label.name) }
+        not_matching_labels.each {|label| expect(page).not_to have_selector('.labels-table > tbody > tr > td > .task_label', text: label.name) }
+      end
+    end
+
+    context 'when merging labels' do
+      let!(:labels) { create_list(:label, 5) }
+      let(:new_name) { 'new label name' }
+
+      context 'when only renaming one label' do
+        subject(:renaming_action) do
+          visit(labels_path)
+          wait_for_ajax
+          find('.labels-table > tbody > tr > td > .task_label', text: labels[1].name).click
+          fill_in('merge-labels-input', with: new_name)
+          accept_prompt do
+            click_button('merge-labels-button')
+          end
+          wait_for_ajax
+        end
+
+        it 'renames the label' do
+          renaming_action
+          expect(Label.find_by(name: new_name)).not_to be_nil
+        end
```

Comment:
Since the merge action changed in the new commit, this test example has now changed too. It now checks for a specific label if the name has changed so the test works even if there is another label with the same name before the renaming_action. This is still a case that I did not consider though. What should happen if after a renaming_action two labels have the same name? Is that acceptable and the responsibility of the admins or should this be prevented? My intuition would be to let the admins handle this. What do you think?

Reply:
I would actually argue that Rails can handle this for us 😁. Since we are assuming that the name is unique (for example, when assigning labels), we should also ensure that they are unique.

What about adding the following to the name validation in the `label.rb` model?

```rb
, uniqueness: {case_sensitive: false}
```

Then, we don't need to consider something manually, but still have the desired result. 
<hr>

Otherwise, the new test looks fine and removes the issue I saw before 👍

Reply:
I will add a test for that too.

Reply:
I just noticed there is a case now where merging labels will fail even though it is supposed to work. If you select two labels a,b and try to merge them with new name 'b' the merge will fail. This is because #merge first tries to rename a to b but this fails because b already exists.
(If you select b,a and rename to b it works as expected)
One way to solve this would be to switch the order of destroying the old labels and updating the merged label:
```
ApplicationRecord.transaction do
  tasks = Task.where(id: TaskLabel.select(:task_id).where(label_id: label_ids))
  Label.destroy(label_ids[1..])
  Label.find(label_ids.first).update!(name: params[:new_label_name], tasks:)
rescue ActiveRecord::ActiveRecordError => e
  render json: e.message, status: :unprocessable_entity
  raise ActiveRecord::Rollback
end
```

Reply:
Ah, that's a good finding. Yes, changing the order sounds reasonable (keep in mind that the transaction guards us against potential errors, so either the merging completes successfully or nothing gets changed).